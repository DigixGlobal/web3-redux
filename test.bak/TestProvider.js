import React, { Component } from 'react';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware } from 'redux';
import { combineReducers } from 'redux-immutable';
import thunk from 'redux-thunk';

import middleware from '../src/middleware';
import reducer from '../src/reducer';

import Web3ConnectTest from './Web3ConnectTest';

const createStoreWithMiddleware = applyMiddleware(thunk, middleware)(createStore);
const reducers = combineReducers({ web3Redux: reducer });
const store = createStoreWithMiddleware(reducers);

export default class TestProvider extends Component {
  render() {
    return (
      <Provider store={store}>
        <Web3ConnectTest />
      </Provider>
    );
  }
}
