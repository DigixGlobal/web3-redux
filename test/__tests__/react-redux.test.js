import React from 'react';
import TestProvider from '../TestProvider';
import renderer from 'react-test-renderer';

import testContract from '../SimpleNameRegistry.sol';

const testAddress = '0x1233000000000000000000000000000000000001';
const gas = 3000000;
let from;
let contractAddress;

const instance = renderer.create(<TestProvider />);
const { triggerMethod } = instance.toJSON().props;

function snapshotTest(getter) {
  if (getter) {
    triggerMethod((context) => context.setData(typeof getter === 'function' && getter(context) || getter));
  }
  expect(instance.toJSON()).toMatchSnapshot();
}

it('initializes correctly', () => {
  snapshotTest(({ props }) => [
    props,
    props.networks,
    props.networks.default,
    props.networks.default.web3,
    props.networks.default.web3.eth,
    props.networks.default.web3.version,
    props.networks.default.web3.net,
  ].map(o => Object.keys(o)));
});

it('gets the correct web3 method values', (done) => {
  // TODO test all values
  triggerMethod(({ props }) => {
    Promise.all([
      props.networks.default.web3.version.getNode(),
      props.networks.default.web3.eth.getBalance(testAddress),
      props.networks.default.web3.eth.getCoinbase().then(res => { from = res; }),
    ])
    .then(() => snapshotTest([
      props.networks.default.web3.version.node(),
      props.networks.default.web3.eth.balance(testAddress),
    ]))
    .then(done);
  });
});

it('deploys contracts', (done) => {
  triggerMethod(({ props }) => {
    const { eth } = props.networks.default.web3;
    const contract = eth.contract(testContract.abi);
    contract.new({ data: testContract.unlinked_binary, from, gas })
    .then((deployed) => {
      snapshotTest(() => Object.keys(deployed));
      contractAddress = deployed.address;
    })
    .then(done);
  });
});

it('contract methods work', (done) => {
  triggerMethod(({ props }) => {
    const { eth } = props.networks.default.web3;
    const contract = eth.contract(testContract.abi);
    const contractInstance = contract.at(contractAddress);
    contractInstance.register.transaction('test', testAddress, { from, gas })
    .then((txHash) => {
      return eth.waitForMined(txHash);
    })
    .then(() => {
      return Promise.all([
        contractInstance.names.call('donate'),
        contractInstance.names.call('test'),
      ]);
    })
    .then(() => {
      snapshotTest([
        contractInstance.names('donate'),
        contractInstance.names('test'),
      ]);
    })
    .then(done);
  });
});
