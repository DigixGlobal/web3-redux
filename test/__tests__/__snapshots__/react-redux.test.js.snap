exports[`test contract methods work 1`] = `
<div
  triggerMethod={[Function]}>
  \"{\\\"test\\\":[\\\"0x8ff77a6a522211faf71242f0ddf61e1645045549\\\",\\\"0x1233000000000000000000000000000000000001\\\"]}\"
</div>
`;

exports[`test deploys contracts 1`] = `
<div
  triggerMethod={[Function]}>
  \"{\\\"test\\\":[\\\"_eth\\\",\\\"transactionHash\\\",\\\"address\\\",\\\"abi\\\",\\\"names\\\",\\\"register\\\",\\\"allEvents\\\"]}\"
</div>
`;

exports[`test gets the correct web3 method values 1`] = `
<div
  triggerMethod={[Function]}>
  \"{\\\"test\\\":[\\\"EthereumJS TestRPC/v3.0.3/ethereum-js\\\",\\\"0\\\"]}\"
</div>
`;

exports[`test initializes correctly 1`] = `
<div
  triggerMethod={[Function]}>
  \"{\\\"test\\\":[[\\\"web3ReduxStore\\\",\\\"networks\\\",\\\"status\\\"],[\\\"default\\\"],[\\\"rawWeb3\\\",\\\"web3\\\"],[\\\"net\\\",\\\"version\\\",\\\"eth\\\"],[\\\"getBalance\\\",\\\"balance\\\",\\\"getCode\\\",\\\"code\\\",\\\"getTransactionCount\\\",\\\"transactionCount\\\",\\\"getStorageAt\\\",\\\"storageAt\\\",\\\"getSyncing\\\",\\\"syncing\\\",\\\"getCoinbase\\\",\\\"coinbase\\\",\\\"getMining\\\",\\\"mining\\\",\\\"getHashrate\\\",\\\"hashrate\\\",\\\"getGasPrice\\\",\\\"gasPrice\\\",\\\"getAccounts\\\",\\\"accounts\\\",\\\"getBlockNumber\\\",\\\"blockNumber\\\",\\\"getBlock\\\",\\\"block\\\",\\\"getBlockTransactionCount\\\",\\\"blockTransactionCount\\\",\\\"getUncle\\\",\\\"uncle\\\",\\\"getTransactionFromBlock\\\",\\\"transactionFromBlock\\\",\\\"getTransaction\\\",\\\"transaction\\\",\\\"getTransactionReceipt\\\",\\\"transactionReceipt\\\",\\\"sendTransaction\\\",\\\"sendRawTransaction\\\",\\\"waitForMined\\\",\\\"contract\\\"],[\\\"getNode\\\",\\\"node\\\",\\\"getNetwork\\\",\\\"network\\\",\\\"getEthereum\\\",\\\"ethereum\\\",\\\"getWhisper\\\",\\\"whisper\\\"],[\\\"getListening\\\",\\\"listening\\\",\\\"getPeerCount\\\",\\\"peerCount\\\"]]}\"
</div>
`;
